coderd/templates.go

```diff
@@ -841,7 +841,17 @@ func (api *API) patchTemplateMeta(rw http.ResponseWriter, r *http.Request) {
 		return nil
 	}, nil)
 	if err != nil {
-		httpapi.InternalServerError(rw, err)
+		if database.IsUniqueViolation(err) {
+			httpapi.Write(ctx, rw, http.StatusConflict, codersdk.Response{
+				Message: fmt.Sprintf("Template %q already exists.", req.Name),
```

Comment:
Definitely out of scope of this PR, but it might be good to have some centralised mapping of database violations to human readable error messages.